cmd/nanotube/main.go

```diff
@@ -75,21 +94,41 @@ func main() {
 
 	// SIGTERM gracefully terminates with timeout
 	// SIGKILL terminates immediately
+	// SIGUSR2 triggers zero-downtime restart
 	sgn := make(chan os.Signal, 1)
-	signal.Notify(sgn, os.Interrupt, syscall.SIGTERM)
-	<-sgn
+	signal.Notify(sgn, os.Interrupt, syscall.SIGTERM, syscall.SIGUSR2)
 
-	// start termination sequence
-	close(stop)
+	for {
+		s := <-sgn
 
-	select {
-	case <-time.After(time.Second * time.Duration(cfg.TermTimeoutSec)):
-		err = lg.Sync()
-		if err != nil {
-			fmt.Fprintf(os.Stderr, "error when syncing logger: %v", err)
+		if s == syscall.SIGUSR2 {
+			lg.Info("Reload: Got signal for reload. Checking config.")
+			_, _, _, _, _, err = loadBuildRegister(cfgPath, lg)
+			if err != nil {
+				lg.Error("Reload: Cannot reload: config invalid", zap.Error(err))
```

Comment:
Thanks for the suggestion. I have thought about that too. Here's a bit of detail.

We need to figure out a way to signal errors. IMO, we should do this via logs because errors are singular events. Thus, I did not make a suggested metric on purpose. I will think about this a bit more in the future and work it out in more detail.

On top of that, we already have an uptime metric in our setup. By looking at it, we will be able to get a kind of monitoring that is close.